build.gradle.kts

```diff
@@ -0,0 +1,112 @@
+import org.jetbrains.changelog.markdownToHTML
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+fun properties(key: String) = project.findProperty(key).toString()
+
+plugins {
+    // Java support
+    id("java")
+    // Kotlin support
+    id("org.jetbrains.kotlin.jvm") version "1.6.0"
+    // Gradle IntelliJ Plugin
+    id("org.jetbrains.intellij") version "1.3.0"
+    // Gradle Changelog Plugin
+    id("org.jetbrains.changelog") version "1.3.1"
+    // Gradle Qodana Plugin
+    id("org.jetbrains.qodana") version "0.1.13"
+}
+
+group = properties("pluginGroup")
+version = properties("pluginVersion")
+
+// Configure project's dependencies
+repositories {
+    mavenCentral()
+}
+
+// Configure Gradle IntelliJ Plugin - read more: https://github.com/JetBrains/gradle-intellij-plugin
+intellij {
+    pluginName.set(properties("pluginName"))
+    version.set(properties("platformVersion"))
+    type.set(properties("platformType"))
+
+    // Plugin Dependencies. Uses `platformPlugins` property from the gradle.properties file.
+    plugins.set(properties("platformPlugins").split(',').map(String::trim).filter(String::isNotEmpty))
+}
+
+// Configure Gradle Changelog Plugin - read more: https://github.com/JetBrains/gradle-changelog-plugin
+changelog {
+    version.set(properties("pluginVersion"))
+    groups.set(emptyList())
+}
+
+// Configure Gradle Qodana Plugin - read more: https://github.com/JetBrains/gradle-qodana-plugin
+qodana {
```

Comment:
This is what came with the newer plugin generation. If I'm fully honest is that I only know that it can run IntelliJs code inspections (probably for CI). I think I'll jump into it and try it out

Reply:
Yeah that's what it is. I've now put it into the Github actions for this project, and it annotates PR's with the issues (Somewhat like sonar). It's a tad bit slow (10 minute setup), but it's nice to have IntelliJ's inspections in the pipeline. 

![image](https://user-images.githubusercontent.com/3278878/152593852-2726af70-a9dd-4fd0-81f5-938ed02cdca8.png)